homeassistant/components/technove/binary_sensor.py

```diff
@@ -0,0 +1,103 @@
+"""Support for TechnoVE binary sensor."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from technove import Station as TechnoVEStation
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import TechnoVEDataUpdateCoordinator
+from .entity import TechnoVEEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class TechnoVEBinarySensorDescription(BinarySensorEntityDescription):
+    """Describes TechnoVE binary sensor entity."""
+
+    value_fn: Callable[[TechnoVEStation], bool | None]
+
+
+BINARY_SENSORS = [
+    TechnoVEBinarySensorDescription(
+        key="conflict_in_sharing_config",
+        translation_key="conflict_in_sharing_config",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.conflict_in_sharing_config,
+    ),
+    TechnoVEBinarySensorDescription(
+        key="in_sharing_mode",
+        translation_key="in_sharing_mode",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.in_sharing_mode,
+    ),
+    TechnoVEBinarySensorDescription(
+        key="is_battery_protected",
+        translation_key="is_battery_protected",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.is_battery_protected,
+    ),
+    TechnoVEBinarySensorDescription(
+        key="is_session_active",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=BinarySensorDeviceClass.BATTERY_CHARGING,
+        value_fn=lambda station: station.info.is_session_active,
+    ),
+    TechnoVEBinarySensorDescription(
+        key="is_static_ip",
+        translation_key="is_static_ip",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        value_fn=lambda station: station.info.is_static_ip,
+    ),
+    TechnoVEBinarySensorDescription(
+        key="update_available",
```

Comment:
This one should be the `update` platform

Reply:
Also when you can't install updates remotely? Or know what version you have/can upgrade to?

Reply:
> Also when you can't install updates remotely?

Yes I think so

>Or know what version you have/can upgrade to?

Probably not as we need to know upgradable version to have a proper state. I just assumed we knew if we know an upgrade is available (or perhaps "fake one") as I think it looks nicer for the users to get it in the updates available section i in `settings` rather than just as a `binary_sensor` even if it's not possible to update it from there.

Reply:
Looks like we only know the current version and if it's up to date

https://github.com/Moustachauve/pytechnove/blob/9c89e482f6c69093fb4f919b67a6d70aea6ec7d6/src/technove/models.py#L119-L130

Reply:
Maybe right now but are we sure it can't be returned from the api of the device?

Reply:
For now, I have no way to know what version a user can upgrade to. I also have no idea how to get the next version. Should I just leave this info out of the integration until I have more details?

All I can get from the device API is a bool that tells me if an update is available, but I'm not 100% certain this ever changes (there are not frequent updates so I can't test currently). I think the app does all the logic for updates right now.

Reply:
Let's leave it in then 👍